stripe/src/main/java/com/stripe/android/view/ShippingFlowActivity.java

```diff
@@ -0,0 +1,169 @@
+package com.stripe.android.view;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.view.ViewPager;
+
+import com.stripe.android.R;
+import com.stripe.android.model.Address;
+import com.stripe.android.model.ShippingMethod;
+
+import java.util.ArrayList;
+
+/**
+ * Activity containing a two-part shipping flow that allows users to provide a shipping address
+ * as well as select a shipping method.
+ */
+public class ShippingFlowActivity extends StripeActivity {
+
+    private ViewPager mViewPager;
+    private ShippingFlowPagerAdapter mShippingFlowPagerAdapter;
+
+    static final String EXTRA_SHIPPING_FLOW_CONFIG = "shipping_flow_config";
+
+    public static class IntentBuilder {
+        private ArrayList<String> mHiddenAddressFields;
+        private ArrayList<String> mOptionalAddressFields;
```

Comment:
Any reason these are `ArrayList` items instead of `List`? Also, is it bad for them to be `null` when they get put into the Intent? (I don't know offhand what happens when you do that)

Reply:
there guys are arraylists because there used to not be an ShippingFlowConfig was not originally a parcelable config and intent have a putStringArrayList method that expects an array list. 

good catch -- I'll make it a list!